### PR TITLE
Respect reduced-motion in the customizer

### PR DESCRIPTION
**Pull request by kevin940726**
Respect `prefers-reduced-motion: reduce` in the customizer.

https://user-images.githubusercontent.com/7753001/130192900-a2b47af0-8163-4e2d-a1e9-bfaa8bbb05be.mp4

Test steps:
1. Enable `reduce motion` in your OS settings ([macOS](https://support.apple.com/guide/mac-help/reduce-screen-motion-mchlc03f57a1/mac) for instance)
2. Visit Appearance -> Customize
3. Click around different sections and panels
4. There should be no transition nor animation effects
5. Focus should not lost when expanding sections
6. Everything should work fine when disabling `reduce motion` too

Trac ticket: https://core.trac.wordpress.org/ticket/53542

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/js/_enqueues/wp/customize/controls.js [hunk 1]
```diff
@@ -1264,11 +1266,14 @@
...
 		_animateChangeExpanded: function( completeCallback ) {
-			// Return if CSS transitions are not supported.
-			if ( ! normalizedTransitionendEventName ) {
-				if ( completeCallback ) {
-					completeCallback();
-				}
+			// Return if CSS transitions are not supported or if reduced motion is enabled.
+			if ( ! normalizedTransitionendEventName || isReducedMotion ) {
+				// Schedule the callback until the next tick to prevent focus loss.
+				_.defer( function () {
```
Comment (tellthemachines) [inline src/js/_enqueues/wp/customize/controls.js:1272]: Where was the focus loss occurring?

**Review by tellthemachines — COMMENTED** [review]
This is working well as long as we reload the page after changing system settings 😅 

I'm not sure how much of an issue it is, but if we load the Customizer with "Reduce motion" enabled and then disable it, but don't reload the Customizer, interacting with the sections makes them go partially off-screen:

Screen Shot 2021-08-23 at 11 31 59 am

When doing it the other way around (starting with "Reduce motion" off and then enabling it), everything looks fine but it's impossible to tab into each section with the keyboard.

File: src/js/_enqueues/wp/customize/controls.js (hunk 1 above)
Comment (kevin940726) [reply src/js/_enqueues/wp/customize/controls.js:1272]: It should focus the "Back" button when expanding to a section. That is lost if we call the callback synchronously though.

**Comment by kevin940726** [comment]
@tellthemachines Good catch! I don't think it's too much of an issue but I pushed a fix anyway :).

File: src/wp-admin/css/customize-controls.css [hunk 2]
```diff
@@ -105,6 +105,12 @@ body:not(.ready) #customize-save-button-wrapper .save {
 	height: 100%;
 }
 
+@media screen and (prefers-reduced-motion: reduce) {
```
Comment (getsource) [inline src/wp-admin/css/customize-controls.css:108]: Just wanted to double-check, as I'm not very CSS knowledgable.

The other examples https://github.com/WordPress/wordpress-develop/search?q=prefers-reduced-motion&type=code use `@media (prefers-reduced-motion: reduce)`  or `@media (prefers-reduced-motion)` (in Twenty Twenty and Twenty Twenty-One), both without specifying `screen`.

I'm wondering if it that means it should apply to contexts other than `screen` as well. What do you think?

Comment (kevin940726) [reply src/wp-admin/css/customize-controls.css]: Yeah, I don't think it matters, but couldn't hurt to remove it :)

**Comment by hellofromtonya** [comment]
Committed via https://core.trac.wordpress.org/changeset/51677.

**Pull request closed without merging**